### PR TITLE
#789 Phase 1: Extraction of texts in netlify-cms-core to external file

### PR DESCRIPTION
**Pull request by nahueld**
**Summary**
As explained in #789 , this is the first phase of the 'big plan' to provide localization to the CMS.

In this phase the CMS should look exactly the same, the only difference is that most (see below) of the texts in `netlify-cms-core` have been extracted to an external file, using `react-polyglot` we provide property `t` to components to render the texts with the matching key.

Caveats:
- Not sure if all texts are included, I might have missed some of them, however if this PR is approved the missing texts can be included later on either by me or some other collaborator.
- Other packages under `./packages` have not been analyzed, however it is easy to send `t` down the wire to provide localization, this can be done later on as well or in parallel by other collaborators.

**Test plan**
- Visual inspection of the app.
- All current tests are still passing.
- If this PR is approved (or once it is approved) we can write unit tests in the components under `netlify-cms-core`, in case you wonder why not writing tests in advance?, I'd rather wait for someone to approve the current approach before getting into the test creation frenzy.

**A picture of a cute animal (not mandatory but encouraged)**
Baby alpaca
alt

### CODE REVIEW

**Comment by nahueld** [comment]
Looks like the machine ran out of space.

**Comment by erquhart** [comment]
@nahueld awesome awesome awesome!!! 👍 👍 

Will review soon!

**Comment by verythorough** [comment]
Deploy preview for *netlify-cms-www* ready!

Built with commit 5cd50f67093a05b5d070398b743d75465d3a81a7

https://deploy-preview-1708--netlify-cms-www.netlify.com

**Comment by verythorough** [comment]
Deploy preview for *cms-demo* ready!

Built with commit 5cd50f67093a05b5d070398b743d75465d3a81a7

https://deploy-preview-1708--cms-demo.netlify.com

File: packages/netlify-cms-core/src/components/App/App.js [hunk 1]
```diff
@@ -88,14 +90,14 @@ class App extends React.Component {
     this.props.loginUser(credentials);
   }
 
-  authenticating() {
+  authenticating(t) {
```
Comment (erquhart) [inline packages/netlify-cms-core/src/components/App/App.js]: No need to pass `t` to the function, just use `this.props.t`.

File: packages/netlify-cms-core/src/components/App/App.js [hunk 2]
```diff
@@ -63,17 +64,18 @@ class App extends React.Component {
     useMediaLibrary: PropTypes.bool,
     openMediaLibrary: PropTypes.func.isRequired,
     showMediaButton: PropTypes.bool,
+    t: PropTypes.func.isRequired,
   };
 
-  static configError(config) {
+  static configError(t, config) {
```
Comment (erquhart) [inline packages/netlify-cms-core/src/components/App/App.js]: Use props instead of passing `t`.

**Comment by erquhart** [comment]
@nahueld I added a couple of comments about passing `t` around. Overall I'm good with the choice to use polyglot and the changes in general, but I'd like to improve the phrases object key names and structure before we merge.

If anyone else wants to jump in on that I'm happy to step out!

Really excited to move this forward! Thank you @nahueld!!

**Comment by nahueld** [comment]
@erquhart thanks for the feedback, will update the PR with the requested changes as soon as possible. I also invited you to the fork so you can change the keys.

**Comment by erquhart** [comment]
Thanks! No need for the invite, I'm able to push commits to your branch as long as you have this PR open.

**Comment by nahueld** [comment]
@erquhart cool, I didn't know that.

Would you find helpful some unit tests using enzyme in the core package?

**Comment by nahueld** [comment]
@erquhart Any updates on this? I can help you to rename the keys if you want, just let me know what are the changes required.

**Comment by nahueld** [comment]
@erquhart Any updates on this?

File: packages/netlify-cms-core/src/actions/editorialWorkflow.js [hunk 3]
```diff
@@ -244,7 +244,10 @@ export function loadUnpublishedEntry(collection, slug) {
         } else {
           dispatch(
             notifSend({
-              message: `Error loading entry: ${error}`,
+              message: {
+                key: 'ui.toast.onFailToLoadEntries',
+                details: error,
+              },
```
Comment (erquhart) [inline packages/netlify-cms-core/src/actions/editorialWorkflow.js]: I can't find these options in the redux-notifications docs, do you have a link for that?

**Comment by abumalick** [comment]
Thanks a lot for working on this feature, we really need translation of the interface.

Why didn't you use react-intl for this?
It is widely used in the react world..

**Comment by erquhart** [comment]
I'll be requesting reviews on PR's in tomorrow's community meeting. If we don't get any more reviews from that I'll go ahead and merge.

File: packages/netlify-cms-core/src/actions/editorialWorkflow.js (hunk 3 above)
Comment (nahueld) [reply packages/netlify-cms-core/src/actions/editorialWorkflow.js]: no, these are custom properties we send to the view.

**Comment by nahueld** [comment]
> Thanks a lot for working on this feature, we really need translation of the interface.
> 
> Why didn't you use react-intl for this?
> It is widely used in the react world..

It's , however is much more complicated to implement, Polyglot is enough for our needs and is much more simpler.

**Comment by nahueld** [comment]
> @nahueld I'll be requesting reviews on PR's in tomorrow's community meeting. If we don't get any more reviews from that I'll go ahead and merge.

I've been very busy the last couple of months, still interested in finishing this.

**Pull request merged**